core/unit_test/TestAtomicViews.hpp

```diff
@@ -1065,11 +1065,8 @@ T AndEqualAtomicViewCheck(const int64_t input_length) {
   const int64_t N = input_length;
   T result[2]     = {1};
   for (int64_t i = 0; i < N; ++i) {
-    if (N % 2 == 0) {
-      result[0] &= (T)i;
-    } else {
-      result[1] &= (T)i;
-    }
+    int64_t idx = N % 2;
+    result[idx] &= (T)i;
```

Comment:
```suggestion
    result[N % 2] &= (T)i;
```

Could simplify even further

Reply:
No I found a thing the compiler accepts I leave it at that ...